Sources/VLCActionSheet.swift

```diff
@@ -0,0 +1,422 @@
+            icon.heightAnchor.constraint(equalToConstant: 25),
+            icon.widthAnchor.constraint(equalTo: icon.heightAnchor),
+
+            name.trailingAnchor.constraint(equalTo: stackView.trailingAnchor),
+            name.centerYAnchor.constraint(equalTo: stackView.centerYAnchor),
+
+            stackView.leadingAnchor.constraint(equalTo: guide.leadingAnchor, constant: 20),
+            stackView.trailingAnchor.constraint(equalTo: guide.trailingAnchor, constant: -20),
+            stackView.heightAnchor.constraint(equalTo: heightAnchor),
+            stackView.topAnchor.constraint(equalTo: topAnchor)
+            ])
+    }
+}
+
+// MARK: VLCActionSheetSectionHeader
+class VLCActionSheetSectionHeader: UIView {
+
+    static let identifier = "VLCActionSheetSectionHeader"
+
+    let title: UILabel = {
+        let title = UILabel()
+        title.font = UIFont.systemFont(ofSize: 16)
+        title.translatesAutoresizingMaskIntoConstraints = false
+        return title
+    }()
+
+    let separator: UIView = {
+        let separator = UIView()
+        separator.backgroundColor = .lightGray
+        separator.translatesAutoresizingMaskIntoConstraints = false
+        return separator
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+        setupSeparator()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupView()
+        setupSeparator()
+    }
+
+    fileprivate func setupSeparator() {
+        addSubview(separator)
+        NSLayoutConstraint.activate([
+            separator.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 20),
+            separator.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -20),
+            separator.heightAnchor.constraint(equalToConstant: 0.5),
+            separator.topAnchor.constraint(equalTo: bottomAnchor, constant: -1)
+            ])
+    }
+
+    fileprivate func setupView() {
+        addSubview(title)
+        NSLayoutConstraint.activate([
+            title.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 20),
+            title.centerYAnchor.constraint(equalTo: centerYAnchor),
+            title.centerXAnchor.constraint(equalTo: centerXAnchor),
+            title.topAnchor.constraint(equalTo: topAnchor)
+            ])
+    }
+}
+
+@objc protocol VLCActionSheetDataSource: class {
+    @objc func numberOfRows() -> Int
+    @objc func actionSheet(collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell
+}
+
+@objc protocol VLCActionSheetDelegate: class {
+    @objc optional func headerViewTitle() -> String?
+    @objc func itemAtIndexPath(_ indexPath: IndexPath) -> Any?
+    @objc func actionSheet(collectionView: UICollectionView, didSelectItem item: Any, At indexPath: IndexPath)
+}
+
+// MARK: VLCActionSheet
+class VLCActionSheet: UIViewController {
+
+    private let cellHeight: CGFloat = 50
```

Comment:
idem.

Reply:
add a todo that these cells need to be dynamic in height in case someone uses big accessibilitytext